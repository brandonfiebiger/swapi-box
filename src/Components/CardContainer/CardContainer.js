import React from 'react';
import './CardContainer.css';

import Card from '../Card/Card'

const CardContainer = ({ category, toggleFavorites, favorites }) => {
  const displayedCards = category.map( (facts, index) => {
    const selected = favorites.includes(facts)
    return <Card key={category.name + index} facts={facts} toggleFavorites={toggleFavorites} selected={selected} />
  });

  return (
    <section>
      {displayedCards}
    </section>
)
}

export default CardContainer;