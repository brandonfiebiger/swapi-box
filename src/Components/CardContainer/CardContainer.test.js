import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';

import CardContainer from './CardContainer';

describe('CardContainer', () => {

  let wrapper;
  let infoCards;
  beforeEach(() => {
    infoCards = [];
    wrapper = shallow(<CardContainer infoCards={infoCards}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when one card is in infoCards array', () => {
    infoCards = [{}];
    expect(wrapper).toMatchSnapshot();  
  })

  it('should match the snapshot when multiple cards are in infoCards array', () => {
    infoCards = [{}, {}];
    expect(wrapper).toMatchSnapshot();  
  })
})