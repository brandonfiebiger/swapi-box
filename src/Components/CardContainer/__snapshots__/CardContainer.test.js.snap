// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `undefined`;

exports[`CardContainer should match the snapshot when multiple cards are in infoCards array 1`] = `undefined`;

exports[`CardContainer should match the snapshot when one card is in infoCards array 1`] = `undefined`;
